flake.nix

```diff
@@ -0,0 +1,72 @@
+{
+  description = "Minimalist CLI email client";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    utils.url = "github:numtide/flake-utils";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    crate2nix = {
+      url = "github:balsoft/crate2nix/tools-nix-version-comparison";
+      flake = false;
+    };
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+  };
+
+  outputs = { self, nixpkgs, utils, rust-overlay, crate2nix, ... }:
+    utils.lib.eachDefaultSystem
+      (system:
+       let 
+          name = "himalaya";
+          pkgs = import nixpkgs { 
+            inherit system; 
+            overlays = [ 
+              rust-overlay.overlay
+              (self: super: {
+                # Because rust-overlay bundles multiple rust packages into one
+                # derivation, specify that mega-bundle here, so that naersk
+                # will use them automatically.
```

Comment:
This is not using naersk anymore

Reply:
Right, I'll remove it. Do you think this "hack" is the correct approach though?

Reply:
It's alright, but note that rust-overlay currently doesn't play too well with newer Nix versions, because of changes to how fixed-hash derivations are handled. In particular, this happens if you try to build on nix 2.4pre without first building on older Nix:

```
error: hash mismatch in fixed-output derivation '/nix/store/az96g89ncj19m2mqgnic7ikkmm0bcgfh-cargo-1.51.0-x86_64-unknown-linux-gnu.tar.xz.drv':
         specified: sha256-LFV+RIwUXtdzuq4NZTNEmUe7Ew+PiioIdtCPVbdKMT8=
            got:    sha256-zx+SwH7GK9W0E9BMbYiouohc36OoYOnmZ/eHS3/u4BI=
```

Reply:
Wait, so this would be broken in a fresh machine with Flakes enabled (as that will use 2.4pre)?

Reply:
Yep, unfortunately so. You can add a note about it somewhere in the readme. It can be fixed with `nix shell nixpkgs#nix -c nix-build`